src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -1400,45 +1412,50 @@ dnnl::post_ops program_node::try_optimize_post_ops(dnnl::post_ops& p_ops, const
         GPU_DEBUG_TRACE << "    " << i << ": " << cur_post_ops[i].op_type << std::endl;
     GPU_DEBUG_TRACE << "------------------------------------------------" << std::endl;
 
-    add_onednn_fused_primitives(cur_post_ops);
-
     return optimized_p_ops;
 }
 
-
-void program_node::init_onednn_primitive_attributes() {
-    const std::vector<fused_primitive_desc>& cldnn_post_ops = get_fused_primitives();
-    auto attrs = std::make_shared<dnnl::primitive_attr>();
+void cldnn::create_onednn_primitive_attributes(
+        const cldnn::program_node& node,
+        const std::vector<fused_primitive_desc>& cldnn_post_ops,
+        std::shared_ptr<dnnl::primitive_attr>& attrs,
+        std::vector<fused_primitive_desc_onednn>& fused_ops,
+        kernel_impl_params* impl_params = nullptr) {
```

Comment:
could you explain why you moved this function out of program_node? I reviewed the code but I could not understand the motivation..

Reply:
move into the program_node as method.